src/gtk/misc-gtk.c

```diff
@@ -246,10 +246,64 @@ set_menu_sensitive (gftp_window_data * wdata, char *path, int sensitive)
 }
 
 
+char *
+report_list_info(gftp_window_data * wdata)
+{
+  GList *templist;
+  int  filenum = 0, dirnum = 0, linknum = 0;
+  off_t filesize = 0;
+  gftp_file *fle;
+  char listreport[256] = "";
+  char files_str[50] = "";
+  char links_str[50] = "";
+  char dirs_str[50] = "";
+  char size_str[50] = "";
+
+  if(!wdata || !wdata->files)
+     return NULL;
+
+  templist=wdata->files;
+  while (templist != NULL)
+  {
+      fle=(gftp_file*)(templist->data);
+      if(fle->shown && strcmp(fle->file,".."))
+      {
+        if(S_ISLNK (fle->st_mode))      linknum++;
+        else if(S_ISDIR (fle->st_mode)) dirnum++;
+        else {
+           filenum++;
+           filesize += fle->size;
+        }
+      }
+      templist = templist->next;
+  }
+  if (!filenum && !dirnum && !linknum) {
+    return NULL;
+  }
+  if(filenum) {
+     snprintf(files_str, sizeof(files_str),  _(" Files: %d"), filenum);
+  }
+  if(dirnum) {
+     snprintf(dirs_str, sizeof(dirs_str), _(" Dirs: %d"), dirnum);
+  }
+  if(linknum) {
+     snprintf(links_str, sizeof(links_str), _(" Dirs: %d"), dirnum);
+  }
+  if(filesize) {
+     char *ofstr = gftp_format_file_size(filesize);
+     snprintf(size_str, sizeof(size_str), _(" Size: %s"), ofstr);
+     g_free (ofstr);
+  }
+  snprintf(listreport, sizeof(listreport), " [%s%s%s%s ]",
```

Comment:
The messages can't be concatenated together like this due to internationization reasons. For example, this assumes a left to right language, but it won't work for a right to left language such as Arabic. I suggest doing just a single snprintf with all of the fields on it if possible.

That said, my old code where report_list_info() is called chains together a message in the same fashion so it's up to you if you want to leave it as is. I just wanted to make you aware of this issue though.

Reply:
I think I'll leave it as is for now.